src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -546,6 +562,226 @@ def _visit_deref(self, node: gtir.FunCall) -> DataExpr:
+
+            else_body = dace.sdfg.state.ControlFlowRegion("else_body", sdfg=nsdfg)
+            fstate = else_body.add_state("false_branch", is_start_block=True)
+            if_region.add_branch(dace.sdfg.state.CodeBlock("not (__cond)"), else_body)
+
+        else:
+            entry_state = nsdfg.add_state("entry", is_start_block=True)
+            tstate = nsdfg.add_state("true_branch")
+            nsdfg.add_edge(entry_state, tstate, dace.InterstateEdge(condition="__cond"))
+            fstate = nsdfg.add_state("false_branch")
+            nsdfg.add_edge(entry_state, fstate, dace.InterstateEdge(condition="not (__cond)"))
+
+        nsdfg_symbol_mapping = {}
+        input_memlets: dict[str, tuple[dace.nodes.AccessNode, Optional[dace_subsets.Range]]] = {}
+
+        if isinstance(condition_value, SymbolExpr):
+            nsdfg.add_symbol("__cond", dace.dtypes.bool)
+            nsdfg_symbol_mapping["__cond"] = condition_value.value
+        else:
+            nsdfg.add_scalar("__cond", dace.dtypes.bool)
+            if isinstance(condition_value, ValueExpr):
+                input_memlets["__cond"] = (
+                    condition_value.dc_node,
+                    dace_subsets.Range.from_string("0"),
+                )
+            else:
+                assert isinstance(condition_value, MemletExpr)
+                input_memlets["__cond"] = (condition_value.dc_node, condition_value.subset)
+                nsdfg_symbol_mapping.update(
+                    {sym: sym for sym in condition_value.subset.free_symbols}
+                )
+
+        def visit_branch(
+            state: dace.SDFGState, expr: gtir.Expr
+        ) -> tuple[
+            list[DataflowInputEdge],
+            DataflowOutputEdge | tuple[DataflowOutputEdge | tuple[Any, ...], ...],
+        ]:
+            assert state in nsdfg.states()
+
+            def visit_arg(arg: IteratorExpr | DataExpr) -> IteratorExpr | ValueExpr:
+                if isinstance(arg, IteratorExpr):
+                    arg_node = arg.field
+                    arg_desc = arg_node.desc(self.sdfg)
+                    arg_subset = dace_subsets.Range.from_array(arg_desc)
+
+                else:
+                    assert isinstance(arg, (MemletExpr, ValueExpr))
+                    arg_node = arg.dc_node
+                    if isinstance(arg, MemletExpr):
+                        assert set(arg.subset.size()) == {1}
+                        arg_desc = dace.data.Scalar(arg_node.desc(self.sdfg).dtype)
+                        arg_subset = arg.subset
+                    else:
+                        arg_desc = arg_node.desc(self.sdfg)
+                        arg_subset = None
+
+                arg_data = arg_node.data
+                # SDFG data containers with name prefix '__tmp' are expected to be transients
+                inner_data = (
+                    arg_data.replace("__tmp", "__input")
+                    if arg_data.startswith("__tmp")
```

Comment:
Just a very picky comment.
You explicitly check if the variable starts with `__tmp`, but replace will replace every occurence of `__tmp`.